src/default-directory-searcher.coffee

```diff
@@ -89,6 +91,8 @@ class DefaultDirectorySearcher
           reject()
     return {
       then: promise.then.bind(promise)
+      catch: promise.catch.bind(promise)
+      directorySearch: directorySearch
```

Comment:
I think it'd be kinda nice to avoid adding the `directorySearch` to the return value. It's because you need to spy on `terminate()` for the specs, right? What do you think about using a spy on `Task.prototype` instead? Maybe just:

```coffee
spyOn(Task::, 'start').andCallThrough()
spyOn(Task::, 'terminate').andCallThrough()

# ...

expect(Task::start.callCount).toBe(1)
expect(Task::terminate.callCount).toBe(1)
```